orderer/common/cluster/block_fetcher_test.go

```diff
@@ -708,3 +956,55 @@ func TestAttestationPullerPullAttestationsEmptyEndpoint(t *testing.T) {
 	ap.Close()
 	dialer.assertAllConnectionsClosed(t)
 }
+
+type failingDialer struct {
+	countingDialer
+}
+
+func (d *failingDialer) Dial(address cluster.EndpointCriteria) (*grpc.ClientConn, error) {
+	return nil, errors.New("failed to dial connection")
+}
+
+func newFailingCountingDialer() *failingDialer {
+	gRPCBalancerLock.Lock()
```

Comment:
i don't understand why you need a lock here? it's all a single thread no?

Reply:
not needed. just a copy paste. will remove it.